packages/arb-util/value/tuple.go

```diff
@@ -270,27 +258,26 @@ func (tv TupleValue) internalHash() (common.Hash, common.Hash) {
 		hashing.Bytes32ArrayEncoded(hashes),
 	)
 
-	finalHash := tv.HashPreImage(firstHash, tv.internalSize())
-	return finalHash, firstHash
+	return firstHash
 }
 
 func (tv TupleValue) Hash() common.Hash {
 	if tv.deferredHashing {
-		hash, preImageHash := tv.internalHash()
-		tv.cachedHash = hash
-		tv.cachedPreImage = preImageHash
+		preImageHash := tv.getPreImage()
+		tv.cachedPreImage = HashPreImage{preImageHash, tv.Size()}
+		tv.cachedHash = tv.cachedPreImage.Hash()
 		tv.deferredHashing = false
 	}
 	return tv.cachedHash
 }
 
 func (tv TupleValue) GetPreImage() HashPreImage {
 	if tv.deferredHashing {
-		hash, preImageHash := tv.internalHash()
-		tv.cachedHash = hash
-		tv.cachedPreImage = preImageHash
+		preImageHash := tv.getPreImage()
+		tv.cachedPreImage = HashPreImage{preImageHash, tv.Size()}
+		tv.cachedHash = tv.cachedPreImage.Hash()
 		tv.deferredHashing = false
 	}
 
-	return HashPreImage{tv.cachedPreImage, tv.Size()}
+	return tv.cachedPreImage
 }
```

Comment:
These functions have a bunch bunch of common code. I think just refactoring to a common private function would be good.